src/audio/eq_fir/eq_fir.c

```diff
@@ -723,40 +703,30 @@ static int eq_fir_copy(struct comp_dev *dev)
 {
 	struct comp_copy_limits cl;
 	struct comp_data *cd = comp_get_drvdata(dev);
-	struct fir_state_32x16 *fir = cd->fir;
-	int nch;
+	int n;
 
 	tracev_eq_with_ids(dev, "eq_fir_copy()");
 
 	/* Get source, sink, number of frames etc. to process. */
 	comp_get_copy_limits(dev, &cl);
 
-	nch = cl.source->channels;
-
-	/* Check if number of frames to process if it is odd. The
-	 * optimized FIR function to process even number of frames
-	 * is lower load than generic version. In that case process
-	 * one frame with generic FIR and the rest with even frames
-	 * number FIR version.
+	/* Process only even number of frames with the FIR function. The
+	 * optimized filter function loads the successive input samples from
+	 * internal delay line with a 64 bit load operation. The any number
+	 * of frames FIR version would break the delay line alignment if called
```

Comment:
Also, what does "The any number of frames FIR version" mean?

Reply:
Maybe "no constraints" version or "odd/even" version?

Reply:
The wording is now "The other odd (or any) number of frames capable FIR version would permanently break the delay line alignment if called with odd number of frames so it can't be used here"

Reply:
looks good now @singalsu